docs/guides/batching.md

```diff
@@ -0,0 +1,183 @@
+|           | `POST`      | 6. `/batch/`                  | Request body     | `multipart/mixed`  | [google.com](https://cloud.google.com/storage/docs/batch)                                        |
+
+### Batching: Sync or Async
+
+REST APIs that support batching can either follow a synchronous or asynchronous style depending on the underlying operation.
+
+The sync style is the most common and is often used for operations that are short-lived i.e. operations that can be completed quickly so that the server can return an immediate response (`200 OK`). In fact, out of the 6 different URL styles shown in the table above, only URL style #5 is asynchronous, the rest are synchronous.
+
+The async style is used in situations where an operation can take a considerable amount of time to complete. The server will process the request asynchronously but will return an immediate response (`202 Accepted`)[^202-Accepted] instead of letting the client wait.
+
+### Receiving an Async Response: Pull or Push
+
+Within the async request style, there are two ways of retrieving a response from the server: pull or push.
+
+In the pull model: the client periodically polls the server to check that the operation has completed successfully, or failed.
+
+In the push model: when the operation is complete, the server pushes the results over an existing subscription with the client such as a web socket (for browser-based clients) or a web hook (for server-based clients).[^delivery]
+
+### Open Data Protocol
+
+URL style #6 in the table above, uses a `Content-type` of `multipart/mixed` which makes it the most flexible way of implementing batching in REST APIs. It allows clients to submit arbitrary operations (multiple _Create_, _Read_, _Update_, and _Delete_ operations, each with its own `Content-type`) in a single request, though most services enforce a limit[^batch-size-limit] in the range of 10-1000 called the batch size. The batch size is the number of sub-requests that can be included in a single request to an endpoint that supports batching.
+
+This style of batching has been made into a standard by Microsoft under the [Open Data Protocol](https://en.wikipedia.org/wiki/Open_Data_Protocol) as the [OData batch processing system](https://www.odata.org/documentation/odata-version-3-0/batch-processing/). Sharepoint Online and Office 365 REST APIs are examples of Microsoft services that support [batching with their REST API](https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/make-batch-requests-with-the-rest-apis) using the OData standard.
+
+Some Google services have adopted the OData syntax, though the semantics differ. Examples include Google [Cloud Storage](https://cloud.google.com/storage/docs/batch), which supports up to 100 requests in a single batch request and Google [Workspace Admin API](https://developers.google.com/admin-sdk/directory/v1/guides/batch), which supports up to 1000 requests in a single batch request.
```

Comment:
feels like too much info on other possible techniques for batching while tailcall supports only one of them. I suggest to focus only on those that are supported by tailcall and describe the main points of that, while other could be replaced by link to additional materials

Reply:
You have a point but while I was writing the article, I realized there were no articles that contain this info in one place. Having an in-depth article that runs through the possible techniques for batching would bring in decent traffic.

Reply:
@meskill I've removed the text you highlighted in the latest commit.